service/history/workflow/workflow_task_state_machine.go

```diff
@@ -472,9 +478,20 @@ func (m *workflowTaskStateMachine) AddWorkflowTaskStartedEvent(
 	// events. That's okay, it doesn't have to be 100% accurate. It just has to be kept
 	// consistent between the started event in history and the event that was sent to the SDK
 	// that resulted in the successful completion.
-	suggestContinueAsNew, historySizeBytes := m.getHistorySizeInfo()
-	if updateReg != nil {
-		suggestContinueAsNew = cmp.Or(suggestContinueAsNew, updateReg.SuggestContinueAsNew())
+	historySizeBytes, suggestContinueAsNewReasons := m.getHistorySizeInfo()
+	suggestContinueAsNew := len(suggestContinueAsNewReasons) > 0
+	if updateReg != nil && updateReg.SuggestContinueAsNew() {
+		suggestContinueAsNew = cmp.Or(suggestContinueAsNew, true)
+		suggestContinueAsNewReasons = append(suggestContinueAsNewReasons, enumspb.SUGGEST_CONTINUE_AS_NEW_REASON_TOO_MANY_UPDATES)
+	}
+
+	if m.ms.GetEffectiveVersioningBehavior() != enumspb.VERSIONING_BEHAVIOR_UNSPECIFIED && targetDeploymentVersion != nil {
+		if currentDeploymentVersion := m.ms.GetEffectiveDeployment(); currentDeploymentVersion != nil &&
```

Comment:
Q: would this work fine if a transition started just at this task for AU wf? expected behavior is no CaN suggestion.

just to double check... I assume `AddWorkflowTaskStartedEvent` is called after transition is started and then `GetEffectiveDeployment` would return the new version which equals target so the condition does not satisfy.

Reply:
Actually the AU workflow _is_ receiving the CaN suggestion on the transition-task, and then once the transition completes, the CaN suggestion is no longer sent to the AU workflow (confirmed by my functional test) -- do we not want that to happen?